### PR TITLE
ci: Add CI/CD and semantic release

### PR DESCRIPTION
**Pull request by rmi22186**
## Instructions
 1. PR target branch should be against `development`
 2. PR title name should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-title-check.yml
 3. PR branch prefix should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-branch-check-name.yml

 ## Summary
We did not have semantic release on this kit.  Since we don't update these kits too often, I wanted to implement this now.  Most of this is pulled and adapted from GA4.  For reference below:
* release.config.js - https://github.com/mparticle-integrations/mparticle-javascript-integration-google-analytics-4/blob/master/packages/GA4Client/release.config.js
* release.yml - https://github.com/mparticle-integrations/mparticle-javascript-integration-google-analytics-4/blob/master/.github/workflows/release.yml

 ## Testing Plan
 - [ ] Was this tested locally? If not, explain why.
It is not possible to test a github action prior to release. However, if there are any failures, we can debugonce this file is in `master`.

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://go.mparticle.com/work/SQDSDKS-5785

### CODE REVIEW

File: test/index.html [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
     </script>
     <!-- change name of XYZ-Kit below to the name of your built kit-->
-    <script src="../build/DoubleClick-Kit.js"></script>
+    <script src="../dist/DoubleClick-Kit.iife.js"></script>
```
Comment (alexs-mparticle) [inline test/index.html:24]: If you're changing the location of this file, shouldn't you also remove the old one?

Comment (rmi22186) [reply test/index.html:24]: Good call.

**Review by alexs-mparticle — CHANGES_REQUESTED** [review]
Looks good. Just remove the old `build` directory and we should be good. Also, Maybe we can introduce a breaking change to verify that tests properly run?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        runs-on: ubuntu-latest
+        needs: confirm-public-repo-master-branch
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v3
+
+            - name: NPM install
+              uses: actions/setup-node@v3
+              with:
+                  node-version: 16.x
```
Comment (alexs-mparticle) [inline .github/workflows/release.yml:28]: Nit: Should we update this to a more recent version of node?

**Review by alexs-mparticle — APPROVED** [review]
:shipit:

**Pull request merged**